src/json-hover.ts

```diff
@@ -0,0 +1,92 @@
+import { type EditorView, Tooltip } from "@codemirror/view";
+import { type Draft, Draft04, JsonSchema } from "json-schema-library";
+import type { JSONSchema7 } from "json-schema";
+
+import { jsonPointerForPosition } from "./utils/jsonPointers";
+import { joinWithOr } from "./utils/formatting";
+
+export type CursorData = { schema?: JsonSchema; pointer: string };
+
+export type FoundCursorData = Required<CursorData>;
+
+export type HoverOptions = {
+  formatHover?: (data: FoundCursorData) => HTMLElement;
+  // todo: handle hover events
+  // onHover?: () => void;
+};
+
+export class JSONHover {
+  private schema: Draft;
+  public constructor(schema: JSONSchema7, private opts?: HoverOptions) {
+    this.schema = new Draft04(schema);
+  }
+  public getDataForCursor(
+    view: EditorView,
+    pos: number,
+    side: -1 | 1
+  ): CursorData | null {
+    const pointer = jsonPointerForPosition(view.state, pos, side);
+
+    if (!pointer) {
+      return null;
+    }
+    const subSchema = this.schema.getSchema(pointer);
+    // doesn't work for oneOf - we need to give it a shape to tell it what kind of schema type to return
```

Comment:
is this still the case? Also can't we give it the current JSON as the shape data?